src/select/dropdown.js

```diff
@@ -6,121 +6,116 @@ LICENSE file in the root directory of this source tree.
-
-import {ICON} from './constants';
-
+import {StyledOptionContent} from './styled-components';
 import {StatefulMenu} from '../menu';
-import {Spinner} from '../spinner';
 import type {DropDownPropsT} from './types';
-import {getOverrides} from '../helpers/overrides';
+import {getOverrides, mergeOverrides} from '../helpers/overrides';
 
-export default function SelectDropDown(props: DropDownPropsT) {
-  const {overrides = {}} = props;
-  const [SelectComponentIcon, selectComponentIconProps] = getOverrides(
-    overrides.SelectComponentIcon,
-    StyledSelectComponentIcon,
-  );
-  const [DropDown, dropDownProps] = getOverrides(
-    overrides.DropDown,
-    StyledDropDown,
-  );
-  const [DropDownItem, dropDownItemProps] = getOverrides(
-    overrides.DropDownItem,
-    StyledDropDownItem,
-  );
-  const [Option, optionProps] = getOverrides(overrides.Option, StyledOption);
-  const [SelectSpinner, selectSpinnerProps] = getOverrides(
-    overrides.SelectSpinner,
-    StyledSelectSpinner,
-  );
-  const {
-    options = [],
-    getOptionLabel,
-    isDropDownOpen,
-    optionsLoaded,
-    selectedOptions,
-    onChange,
-    onItemSelect,
-    type,
-    maxDropdownHeight,
-    multiple,
-  } = props;
-  return isDropDownOpen ? (
-    <StatefulMenu
-      getRequiredItemProps={(option, index) => {
-        return option.disabled
-          ? {
-              onClickCapture: e => e.stopPropagation(),
-            }
-          : {
-              onClick: e => onChange(e, option),
-            };
-      }}
-      overrides={{
-        List: {
-          component: DropDown,
-          style: {maxHeight: maxDropdownHeight},
-          props: {
-            role: 'listbox',
-            'aria-multiselectable': multiple,
-            $type: type,
-            $isOpen: isDropDownOpen,
-            ...dropDownProps,
-          },
-        },
-        Option: {
-          props: {
-            overrides: {
-              ListItem: {
-                component: DropDownItem,
-                props: dropDownItemProps,
+export default class SelectDropDown extends React.Component<DropDownPropsT> {
+  getSharedProps() {
+    const {
+      error,
+      isLoading,
+      multi,
+      required,
+      size,
+      searchable,
+      type,
+    } = this.props;
+    return {
+      $error: error,
+      $isLoading: isLoading,
+      $multi: multi,
+      $required: required,
+      $searchable: searchable,
+      $size: size,
+      $type: type,
+    };
+  }
+  // eslint-disable-next-line flowtype/no-weak-types
+  getItemLabel = (option: {[string]: any}) => {
+    const {getOptionLabel, overrides = {}, value, valueKey} = this.props;
+
+    const [OptionContent, optionContentProps] = getOverrides(
+      overrides.OptionContent,
+      StyledOptionContent,
+    );
+
+    let $selected;
+    if (Array.isArray(value)) {
+      $selected = value.find(
+        selected => selected && selected[valueKey] === option[valueKey],
+      );
```

Comment:
It will, but on the line `61` it's turned into a `boolean`. But I do agree that it's not that nice to have different types assigned to the `$selected`. Changed it here.